mlflow/utils/autologging_utils/__init__.py

```diff
@@ -584,6 +584,9 @@ def disable_discrete_autologging(flavors_to_disable: List[str]) -> None:
         autolog_func.autolog(disable=False)
 
 
+_training_sessions = []
```

Comment:
Why do we need this change?

Reply:
This is an issue captured by spark-ML cross tests.

Some meta estimator fitting like OneVsRest.fit, creating multiple worker threads, and in each worker thread, it calls nested estimator.fit, we need to disable all nested `fit` autologging, and we should not create managed run for these worker threads.

but the managed mlflow run is automatically created by autologging safe-patch routine if it does not find active run in the worker thread: https://github.com/mlflow/mlflow/blob/bcf2bed4992e7af665b2ff17527edca3ea208ef1/mlflow/utils/autologging_utils/safety.py#L179C5-L179C21

so we need to make it skip creating managed mlflow run in this case.